common/authorization/defaultAuthorizer.go

```diff
@@ -43,12 +43,16 @@ func (a *defaultAuthorizer) Authorize(_ context.Context, claims *Claims, target
 	if target.Namespace == "temporal-system" || target.Namespace == "" {
 		return Result{Decision: DecisionAllow}, nil
 	}
-
+	if claims == nil {
+		return Result{Decision: DecisionDeny}, nil
+	}
 	// Check system level permissions
 	if claims.System == RoleAdmin || claims.System == RoleWriter {
 		return Result{Decision: DecisionAllow}, nil
 	}
-
+	if claims.Namespaces == nil {
```

Comment:
technically not necessary since reading from a nil dictionary will always return found = false, so this should already be handled